Type issue with the isolatedModules flag on

# Bug report

## Description / Observed Behavior

I'm getting the following type issues with TypeScript and `swr@1.0.0`:
```
node_modules/swr/dist/types.d.ts:80:6 - error TS2748: Cannot access ambient const enums when the '--isolatedModules' flag is provided.

80     [RevalidateEvent.FOCUS_EVENT]: void;
        ~~~~~~~~~~~~~~~

node_modules/swr/dist/types.d.ts:81:6 - error TS2748: Cannot access ambient const enums when the '--isolatedModules' flag is provided.

81     [RevalidateEvent.RECONNECT_EVENT]: void;
        ~~~~~~~~~~~~~~~

node_modules/swr/dist/types.d.ts:82:6 - error TS2748: Cannot access ambient const enums when the '--isolatedModules' flag is provided.

82     [RevalidateEvent.MUTATE_EVENT]: Promise<boolean>;
        ~~~~~~~~~~~~~~~
```

## Expected Behavior

No type errors.

## Repro Steps / Code Example

My `tsconfig.json` looks like this:
```json
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "baseUrl": "./src",
    "forceConsistentCasingInFileNames": true,
    "incremental": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "pretty": true,
    "resolveJsonModule": true,
    "strict": true,
    "target": "esnext",
    "tsBuildInfoFile": ".cache/tsBuildInfoFile",
    "types": ["jest", "@testing-library/jest-dom"],
    "useUnknownInCatchVariables": false
  },
  "include": ["src/**/*"]
}
```

## Additional Context

SWR version. 1.0.0
Add any other context about the problem here.

This wasn't an issue with the previous versions of swr.

--- a/src/constants/revalidate-events.ts
+++ b/src/constants/revalidate-events.ts
@@ -1,0 +1,3 @@
+export const FOCUS_EVENT = 0
+export const RECONNECT_EVENT = 1
+export const MUTATE_EVENT = 2

--- a/src/types.ts
+++ b/src/types.ts
@@ -1,3 +1,5 @@
+import * as revalidateEvents from './constants/revalidate-events'
+
 export type Fetcher<Data> = (...args: any) => Data | Promise<Data>
 
 // Configuration types that are only used internally, not exposed to the user.
@@ -164,16 +166,15 @@ export type Revalidator = (
   revalidateOpts?: RevalidatorOptions
 ) => Promise<boolean> | void
 
-export const enum RevalidateEvent {
-  FOCUS_EVENT = 0,
-  RECONNECT_EVENT = 1,
-  MUTATE_EVENT = 2
-}
+export type RevalidateEvent =
+  | typeof revalidateEvents.FOCUS_EVENT
+  | typeof revalidateEvents.RECONNECT_EVENT
+  | typeof revalidateEvents.MUTATE_EVENT
 
 type RevalidateCallbackReturnType = {
-  [RevalidateEvent.FOCUS_EVENT]: void
-  [RevalidateEvent.RECONNECT_EVENT]: void
-  [RevalidateEvent.MUTATE_EVENT]: Promise<boolean>
+  [revalidateEvents.FOCUS_EVENT]: void
+  [revalidateEvents.RECONNECT_EVENT]: void
+  [revalidateEvents.MUTATE_EVENT]: Promise<boolean>
 }
 export type RevalidateCallback = <K extends RevalidateEvent>(
   type: K

--- a/src/use-swr.ts
+++ b/src/use-swr.ts
@@ -11,7 +11,7 @@ import { subscribeCallback } from './utils/subscribe-key'
 import { broadcastState } from './utils/broadcast-state'
 import { getTimestamp } from './utils/timestamp'
 import { internalMutate } from './utils/mutate'
-
+import * as revalidateEvents from './constants/revalidate-events'
 import {
   State,
   Fetcher,
@@ -362,7 +362,7 @@ export const useSWRHandler = <Data = any, Error = any>(
     // revalidation from the outside.
     let nextFocusRevalidatedAt = 0
     const onRevalidate = (type: RevalidateEvent) => {
-      if (type === RevalidateEvent.FOCUS_EVENT) {
+      if (type === revalidateEvents.FOCUS_EVENT) {
         const now = Date.now()
         if (
           configRef.current.revalidateOnFocus &&
@@ -372,11 +372,11 @@ export const useSWRHandler = <Data = any, Error = any>(
           nextFocusRevalidatedAt = now + configRef.current.focusThrottleInterval
           softRevalidate()
         }
-      } else if (type === RevalidateEvent.RECONNECT_EVENT) {
+      } else if (type === revalidateEvents.RECONNECT_EVENT) {
         if (configRef.current.revalidateOnReconnect && isActive()) {
           softRevalidate()
         }
-      } else if (type === RevalidateEvent.MUTATE_EVENT) {
+      } else if (type === revalidateEvents.MUTATE_EVENT) {
         return revalidate()
       }
       return

--- a/src/utils/broadcast-state.ts
+++ b/src/utils/broadcast-state.ts
@@ -1,5 +1,6 @@
-import { Broadcaster, RevalidateEvent } from '../types'
+import { Broadcaster } from '../types'
 import { SWRGlobalState, GlobalState } from './global-state'
+import * as revalidateEvents from '../constants/revalidate-events'
 
 export const broadcastState: Broadcaster = (
   cache,
@@ -24,7 +25,7 @@ export const broadcastState: Broadcaster = (
 
   // If we also need to revalidate, only do it for the first hook.
   if (shouldRevalidate && revalidators && revalidators[0]) {
-    return revalidators[0](RevalidateEvent.MUTATE_EVENT).then(() =>
+    return revalidators[0](revalidateEvents.MUTATE_EVENT).then(() =>
       cache.get(key)
     )
   }

--- a/src/utils/cache.ts
+++ b/src/utils/cache.ts
@@ -3,13 +3,14 @@ import { IS_SERVER } from './env'
 import { UNDEFINED, mergeObjects } from './helper'
 import { internalMutate } from './mutate'
 import { SWRGlobalState } from './global-state'
+import * as revalidateEvents from '../constants/revalidate-events'
+import { RevalidateEvent } from '../types'
 
 import {
   Cache,
   ScopedMutator,
   RevalidateCallback,
-  ConfigOptions,
-  RevalidateEvent
+  ConfigOptions
 } from '../types'
 
 function revalidateAllKeys(
@@ -59,14 +60,14 @@ export function initCache<Data = any>(
         revalidateAllKeys.bind(
           UNDEFINED,
           EVENT_REVALIDATORS,
-          RevalidateEvent.FOCUS_EVENT
+          revalidateEvents.FOCUS_EVENT
         )
       )
       opts.initReconnect(
         revalidateAllKeys.bind(
           UNDEFINED,
           EVENT_REVALIDATORS,
-          RevalidateEvent.RECONNECT_EVENT
+          revalidateEvents.RECONNECT_EVENT
         )
       )
     }

--- a/test/use-swr-focus.test.tsx
+++ b/test/use-swr-focus.test.tsx
@@ -195,6 +195,7 @@ describe('useSWR - focus', () => {
     await act(() => sleep(150))
     // trigger revalidation
     await focusWindow()
+    await waitForNextTick()
     await screen.findByText('data: 5')
   })
 

